build.go

```diff
@@ -122,6 +122,18 @@ func Build(
 			return packit.BuildResult{}, err
 		}
 
+		stack, ok := nugetCache.Metadata["stack"].(string)
+		if ok && stack != context.Stack {
+			nugetCache, err = nugetCache.Reset()
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+		}
+
+		nugetCache.Metadata = map[string]interface{}{
+			"stack": context.Stack,
+		}
+
```

Comment:
This will clobber any other elements in the `Metadata` map attached to the nugetCache layer. Is that the desired behaviour?

Reply:
I don't think that there is any other metadata on that later. I have to do this to initialize the map structure.

Reply:
An alternative would be to check if the map is nil before initializing it. In the current implementation, I agree there's no other metadata on the layer. But if we start storing other data on the layer, this will toss it out.

Reply:
I mean I can add a check but I think that this is generally what we do in all of our other buildpacks.